lib/plugins/leetcode.js

```diff
@@ -594,14 +601,36 @@ plugin.githubLogin = function(user, cb) {
       if (resp.statusCode !== 200) {
         return cb('GitHub login failed');
       }
-      _request.get({url: leetcodeUrl}, function(e, resp, body) {
-        const redirectUri = resp.request.uri.href;
-        if (redirectUri !== 'https://leetcode.com/') {
-          return cb('GitHub login failed or GitHub did not link to LeetCode');
+      if (resp.request.uri.href == 'https://github.com/sessions/two-factor') {
+        cb('Your Github are using two-factor authentication');
+        // read two-factor code must be sync.
+        const twoFactorcode = require('prompt-sync')()('Please enter your two-factor code: ');
+        const authenticityTokenTwoFactor = body.match(/name="authenticity_token" value="(.*?)"/);
+        if (authenticityTokenTwoFactor === null) {
+          return cb('Get GitHub two-factor token failed');
         }
-        const cookieData = parseCookie(resp.request.headers.cookie, cb);
-        saveAndGetUser(user, cb, cookieData);
-      });
+        const optionsTwoFactor = {
+          url:     'https://github.com/sessions/two-factor',
+          method:  'POST',
+          headers: {
+            'Content-Type': 'application/x-www-form-urlencoded',
+          },
+          followAllRedirects: true,
+          form:               {
+            'otp':                twoFactorcode,
+            'authenticity_token': authenticityTokenTwoFactor[1],
+            'utf8':               encodeURIComponent('✓'),
+          },
+        };
+        _request(optionsTwoFactor, function(e, resp, body) {
+          if (resp.request.uri.href === 'https://github.com/sessions/two-factor') {
+            return cb('Wrong two-factor code please check');
+          }
+          requestLeetcodeAndSave(_request, leetcodeUrl, user, cb, 'GitHub');
+        });
+      } else {
+        requestLeetcodeAndSave(_request, leetcodeUrl, user, cb, 'GitHub');
```

Comment:
So now we have several login methods:

- LeetCode
- Cookie
- GitHub
- LinkedIn

Could we create a enum for these constants and reference the enum item where we need?